### ISSUE TITLE
The integral of the function is 1, how to write this condition to network?

### ISSUE THREAD

**Issue by Tomandjob** [lululxvi/deepxde#576]
Dear lulu, 
I solved a simple partial differential equation using deepxde, but the predicted result is several orders of magnitude different from the true result. I modified the output of the network, but it did not work.

The integral of **u** over the domain of definition is 1. I think this should be an important condition. How to write it into the network, could you give me some advice？

**Comment by lululxvi** [comment]
See FAQ "Q: I failed to train the network or get the right solution, e.g., large training loss, unbalanced losses."

**Comment by Tomandjob** [comment]
> See FAQ "Q: I failed to train the network or get the right solution, e.g., large training loss, unbalanced losses." and "Q: Implement new losses or constraints."

I reviewed the code. I didn't put the normalization conditions (**∫(u)dx=1**) in the code. I put “0.05*u(x)-1” into the pde. The predicted shape is the same as the actual shape, but the value is several times different. I changed the output, but it didn't work. I referred to Volterra_IDE, but still didn't know how to write the integral into pde.

**Comment by praksharma** [comment]
@Tomandjob  First of all, please do not paste codes with errors. `A=xy` is not valid statement in Python. `return 0.5dy_xx + dA_x` is also not valid. 

Also, when you paste a code use this tool  ![image](https://user-images.githubusercontent.com/70601302/160762179-de46808f-9784-4325-9721-82a94aa51853.png). It helps is preserving the indentation.

I looked into it. This is what I get when I run the same stuff. The predictions are zero.
![5000_epoch](https://user-images.githubusercontent.com/70601302/160767565-9e49436a-63ca-463a-b717-16991ed08545.jpg)
Which is correct. It satisfies the BC as well as the PDE at each point.


Here, is the loss stat:
```sh
Training model...

Step      Train loss                        Test loss                         Test metric
0         [1.58e-02, 2.15e-02, 2.15e-02]    [1.58e-02, 2.15e-02, 2.15e-02]    []  
1000      [9.75e-08, 2.80e-09, 1.14e-10]    [9.75e-08, 2.80e-09, 1.14e-10]    []  
2000      [6.61e-08, 3.77e-09, 1.71e-08]    [6.61e-08, 3.77e-09, 1.71e-08]    []  
3000      [1.98e-07, 4.25e-07, 1.19e-08]    [1.98e-07, 4.25e-07, 1.19e-08]    []  
4000      [5.41e-06, 6.38e-06, 7.66e-06]    [5.41e-06, 6.38e-06, 7.66e-06]    []  
5000      [3.61e-07, 3.65e-07, 3.83e-07]    [3.61e-07, 3.65e-07, 3.83e-07]    []  

Best model at step 2000:
  train loss: 8.69e-08
  test loss: 8.69e-08
  test metric: []

'train' took 83.282071 s
```
DeepXDE always works on well-posed 1D problems. Are you sure the problem is well-posed i.e. the solution is unique?

**Comment by Tomandjob** [comment]
@praksharma  Thank you for your reply. The right codes should be `A=x*y` and `0.5*dy_xx + dA_x`. 

The predicted solution of the neural network is very small and fluctuates around 0.  The code is missing a normalization conditions (∫ydx=1). I put “0.05*y-1” into the pde. The shape of the predicted value curve is the same as the true value curve，but the predicted value is many times larger than the real value.

**Comment by praksharma** [comment]
@Tomandjob I don’t think the implementation of the normalisation condition is correct.

**Comment by Tomandjob** [comment]
@praksharma  Yes, this is an approximate method. I don't know how to define this condition precisely yet.

**Comment by lululxvi** [comment]
@Tomandjob For the normalization, see https://github.com/lululxvi/deepxde/issues/174
